content/blog/2017/2017-01-12-scm-api-2.0-release.adoc

```diff
@@ -0,0 +1,216 @@
+Rather they make queries via the GitHub / BitBucket REST APIs on demand.
+
+The above design decision exposed one of the initial MVP compromises of the SCM API plugin: _Very basic event support_.
+Under the SCM API 1.x model, the only event that an `SCMSource` can signal is _something changed, go look at everything again_.
+When you are accessing an API that only allows 5,000 API calls per hour, performing a full scan of the entire repository just to pick up a change in one branch does not make optimum usage of that 5,000 calls/hour rate limit.
+
+So we decided that perhaps the SCM API and Branch API plugins have left their Minimum Viability Experiment state and the corresponding limitations should be addressed.
+
+=== Enter SCM API 2.0.x and Branch API 2.0.x
+
+We are announcing the SCM API 2.0.x and Branch API 2.0.x release line.
+These plugin releases include:
+
+* https://github.com/jenkinsci/scm-api-plugin/blob/master/docs/implementation.adoc[documentation on how plugin developers are supposed to implement the SCM API]
+* https://github.com/jenkinsci/scm-api-plugin/blob/master/docs/consumer.adoc[documentation on how plugin developers are supposed to consume the SCM API (if they wanted to do something like Branch API but not the same was as Branch API)]
+* https://github.com/jenkinsci/branch-api-plugin/blob/master/docs/implementation.adoc[documentation on how plugin developers are supposed to implement the Branch API to create their own multi-branch project types]
+* https://github.com/jenkinsci/branch-api-plugin/blob/master/docs/user.adoc[generic documentation on how Branch API based project types are intended to work]
+* https://github.com/jenkinsci/scm-api-plugin/blob/master/src/main/java/jenkins/scm/api/SCMEvent.java[a full featured event system that allows implementers to provide fine grained notifications to consumers]
+* https://github.com/jenkinsci/scm-api-plugin/tree/master/src/test/java/jenkins/scm/api[lots] https://github.com/jenkinsci/scm-api-plugin/tree/master/src/test/java/jenkins/scm/impl[and] https://github.com/jenkinsci/branch-api-plugin/tree/master/src/test/java/integration[lots] of new automated tests added
+* https://github.com/jenkinsci/scm-api-plugin/tree/master/src/test/java/jenkins/scm/impl/mock[a mock implementation of the SCM API] to help consumers of the SCM API test their usage.
+
+In addition, we have upgraded the following plugins to include the new fine-grained event support:
+
+* Git Plugin
+* Mercurial Plugin
+
+Ok, that was the good news... Here is the bad news...
+
+We found out that the GitHub Branch Source and BitBucket Branch Source plugins had made invalid assumptions about how to implement the SCM API.
+To be clean, this was not the plugin developers fault: at the time there was no documentation on how to implement the SCM API.
+
+But fixing the issues that we found means that you have to be careful about the matrix of plugin versions that you have installed.
```

Comment:
`s/about the matrix of plugin versions/about which specific combination of plugin versions/`